calendar/modules/listener/http_service.bal

```diff
@@ -79,30 +77,32 @@ service class HttpService {
                 calendar:Time? end = event?.end;
                 if (created is string && updated is string && 'start is calendar:Time && end is calendar:Time) {
                     if (created.substring(0, 19) == updated.substring(0, 19)) {
-                       if (self.isOnNewEventAvailable) {
-                           check callOnNewEventMethod(self.httpService, event);
-                       }
-                    } else  {
-                        if (self.isOnEventUpdateAvailable) {
-                            check callOnEventUpdateMethod(self.httpService, event);
+                        if (self.isOnNewEventAvailable) {
+                            check callOnNewEventMethod(self.httpService, event);
                         }
+                    } else if (self.isOnEventUpdateAvailable) {
+                        check callOnEventUpdateMethod(self.httpService, event);
                     }
-                } else {
-                    if (self.isOnEventDeleteAvailable) {
-                        check callOnEventDeleteMethod(self.httpService, event);
-                    }
+                } else if (self.isOnEventDeleteAvailable) {
+                    check callOnEventDeleteMethod(self.httpService, event);
                 }
+                check caller->respond(res);
             }
+        } else {
+            return error(INVALID_ID_ERROR);
         }
-        return error (INVALID_ID_ERROR);
     }
 
-    isolated function getNextPageToken(calendar:Client httpClient, string calendarId, string? pageToken = ()) returns
-        @tainted string?|error {
+    isolated function getSyncToken(calendar:Client httpClient, string calendarId, string? pageToken = ()) 
```

Comment:
Shall we rename the method to `getNextSyncToken`?

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-googleapis.calendar/commit/4bcb8cdf8fddb87ed2f9a690d8bf9d5c7d47d4df